### PR TITLE
[ANCHOR-686] Add `user_action_required_by`

### PR DESCRIPTION
**Pull request by Ifropc**
### Description

- Add new field `user_action_required_by` to SEP-6/24/31 transactions and related DB entries.
- Added configuration option for SEP-6 and SEP-24 to set initial value on transaction creation.
- Added support for related RPC requests to update this field.
- Added ability to query platform transactions sorted by `user_action_required_by` field

### Context

Implements https://github.com/stellar/stellar-protocol/pull/1484

### Testing

- `./gradlew test`
- New test cases covering all new features

### Documentation

https://stellarorg.atlassian.net/browse/ANCHOR-688
https://stellarorg.atlassian.net/browse/ANCHOR-704

### Known limitations

N/A

### CODE REVIEW

**Review by JiahuiWho — APPROVED** [review]

File: core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt [hunk 1]
```diff
@@ -270,6 +271,26 @@ internal class Sep24ServiceTest {
     assertEquals(withdrawQuote.buyAsset, slotTxn.captured.amountOutAsset)
   }
 
+  @Test
+  fun `test withdraw with user_action_required_by`() {
+    val slotTxn = slot<Sep24Transaction>()
+    val now = Instant.now()
```
Comment (JiahuiWho) [inline core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:277]: nit remove if not used

File: core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt [hunk 2]
```diff
@@ -408,6 +429,26 @@ internal class Sep24ServiceTest {
     assertEquals(depositQuote.sellAsset, slotTxn.captured.amountInAsset)
   }
 
+  @Test
+  fun `test deposit with user_action_required_by`() {
+    val slotTxn = slot<Sep24Transaction>()
+    val now = Instant.now()
```
Comment (JiahuiWho) [inline core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:435]: same

File: core/src/main/java/org/stellar/anchor/sep24/Sep24Helper.java [hunk 3]
```diff
@@ -25,6 +25,7 @@ static void setSharedTransactionResponseFields(TransactionResponse txnR, Sep24Tr
     if (txn.getToAccount() != null) txnR.setTo(txn.getToAccount());
     if (txn.getStartedAt() != null) txnR.setStartedAt(txn.getStartedAt());
     if (txn.getCompletedAt() != null) txnR.setCompletedAt(txn.getCompletedAt());
+    if (txn.getUserActionRequiredBy() != null) txnR.setUserActionRequiredBy(txn.getCompletedAt());
```
Comment (philipliu) [inline core/src/main/java/org/stellar/anchor/sep24/Sep24Helper.java:28]: Should this be set to completed time?

File: core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt [hunk 4]
```diff
@@ -270,6 +271,26 @@ internal class Sep24ServiceTest {
...
+  @Test
+  fun `test withdraw with user_action_required_by`() {
+    val slotTxn = slot<Sep24Transaction>()
+    val now = Instant.now()
+    val deadline = 100L
+    every { txnStore.save(capture(slotTxn)) } returns null
+    every { sep24Config.initialUserDeadlineSeconds } returns deadline
+    sep24Service.withdraw(createTestSep10JwtWithMemo(), createTestTransactionRequest())
+    val dbDeadline = slotTxn.captured.userActionRequiredBy.epochSecond
+    val expectedDeadline = Instant.now().plusSeconds(deadline).epochSecond
```
Comment (philipliu) [inline core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:283]: We really should be injecting a Clock everywhere 😢

File: platform/src/main/java/org/stellar/anchor/platform/rpc/RpcMethodHandler.java [hunk 5]
```diff
@@ -163,6 +164,13 @@ private void updateTransaction(JdbcSepTransaction txn, T request) throws AnchorE
     boolean shouldClearMessageStatus =
         !isErrorStatus(nextStatus) && isErrorStatus(SepTransactionStatus.from(txn.getStatus()));
 
+    txn.setUserActionRequiredBy(null);
+    if (request instanceof SupportsUserActionRequiredBy
+        && ((SupportsUserActionRequiredBy) request).getUserActionRequiredBy() != null) {
+      txn.setUserActionRequiredBy(
```
Comment (philipliu) [inline platform/src/main/java/org/stellar/anchor/platform/rpc/RpcMethodHandler.java:170]: Should we verify that the deadline is not in the past?

Comment (Ifropc) [reply platform/src/main/java/org/stellar/anchor/platform/rpc/RpcMethodHandler.java:170]: Good point

**Review by philipliu — APPROVED** [review]

File: core/src/main/java/org/stellar/anchor/sep24/Sep24Helper.java (hunk 3 above)
Comment (Ifropc) [reply core/src/main/java/org/stellar/anchor/sep24/Sep24Helper.java:28]: Copy paste 🤦🏼

File: core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt (hunk 4 above)
Comment (Ifropc) [reply core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:283]: Created a task for that
https://stellarorg.atlassian.net/browse/ANCHOR-705

**Pull request merged**